### PR TITLE
Docs update for 0.7

### LINKED ISSUE #928: Documentation, test case and example for UDFMeta.coordinates
As a follow-up to #793, we should refine the newly introduced feature. A good candidate to use this in practice is SSB in https://github.com/Ptychography-4-0/ptychography

### LINKED ISSUE #990: Improve Python API docs
- Add a list of functionality: deep links into the reference (at the top) --> postponed via #638 
- [x] Explanation of `functools.partial`
- [x] Integrate the UDF notebook into the UDF docs
- [x] Address https://github.com/Ptychography-4-0/ptychography/issues/44 for LiberTEM (drive-by)

### PR DESCRIPTION
**Pull request by uellue**
Closes #990
Closes #928

## TODO

- [x] Add changelog entry for #1039

## Contributor Checklist:

* [x] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [N/A] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [x] I have added/updated documentation for all user-facing changes
* [N/A] I have added/updated test cases
* [x] I have included the [rebuilt production build of the client](https://libertem.github.io/LiberTEM/contributing.html?#building-the-client) (only if
changes were made to the GUI)

## Reviewer Checklist:

* [x] `/azp run libertem.libertem-data` passed

### CODE REVIEW

**Review by sk1p — APPROVED** [review]
Looks good! I've added two small suggested changes.

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -11,15 +11,113 @@ Changelog
...
+
+* UDF: Consistently use attribute access in :code:`UDF.process_*()`, :code:`UDF.merge()`,
+  :code:`UDF.get_results()` etc. instead of mixing it with :code:`__getitem__()`
+  dict-like access. Also allow non-sliced assignment, for example
+  :code:`self.results.res += frame` (:issue:`1000`, :pr:`1003`).
+* Better choice of :code:`kind='nav'` buffer fill value outside ROI.
+
+  * String : :code:`'n'` -> :code:`''`
+  * bool : :code:`True` -> :code:`False`
+  * integers : smallest possible value -> :code:`0`
```
[the remark this answers is not in the document]
Comment (uellue) [reply docs/source/changelog.rst:89]: Actually, it is the smallest value:

```python
a = np.full(1, fill_value=np.nan, dtype=int)
print(a)

array([-2147483648])
```

Also note how `int` on win64 is an ìnt32` 🙄

[the remark this answers is not in the document]
Comment (sk1p) [reply docs/source/changelog.rst:89]: Yup! Numpy closely follows the data model of the platform here - I kind of dislike the non-specific `dtype=int/float` for general use, I'd always specify the concrete size, if you are not trying to interface with something that really needs the specific `sizeof(int)`.

So what needs fixing - the docs or the implementation?

File: docs/source/udf/advanced.rst [hunk 2]
```diff
@@ -23,36 +27,41 @@ Tiled processing
...
+is the natural block size for K2IS data. That means the tiles will have a shape
+of :code:`(16, 930, 16)`, and processing 16 frames from the data set means
+reading 256 individual tiles.
 
-Loading a tile of this size as float32 data
-still fits comfortably into usual L3 CPU caches (~1MB), and thus enables efficient processing.
-As a comparison, a whole :code:`(1860, 2048)` frame is about 15MB large, and accessing it repeatedly
-means having to load data from the slower main memory.
+Loading a tile of this size as float32 data still fits comfortably into usual L3
+CPU caches (~1MB), and thus enables efficient processing. As a comparison, a
```
Comment (sk1p) [inline docs/source/udf/advanced.rst:56]: Maybe we should clarify that we mean 1MiB per core here?

**Comment by sk1p** [comment]
I think this should also close #928 right?

**Comment by sk1p** [comment]
Thx, merged!

**Pull request merged**